### PR TITLE
ZCS-3608: Added refresh method and searching account instead of loading everytime for some testcases

### PR DESCRIPTION
**Pull request by jiteshsojitra**
ZCS-3608: Added refresh method and searching account instead of loading every time for some testcases

1. Added new menu options and buttons to use O_<menu-name> for account right click
2. Searching account for functional testcases instead of waiting for loading of main accounts (takes lot of time)
3. Fixed issues accordingly and added sleeps at required places
4. Added new method for right click menu options

Lot of old code repaired and may not look fine in phase1, phase2 will have formatting and rest would be easy for us for any such changes. Right now, we can't do together neither we can leave like unorganized and unreliable.

	public AbsPage zListItem(Action action, Button option, String accountEmailAddress) throws HarnessException {
			
		logger.info(myPageName() + " zListItem(" + action + ", " + option + ", " + accountEmailAddress + ")");

		AbsPage page = null;
		String accountLocator, optionLocator = null;
		Boolean accountFound = false;
		
		SleepUtil.sleepMedium();
		
		if (action == Action.A_RIGHTCLICK) {
			
			((AppAdminConsole)MyApplication).zPageSearchResults.zAddSearchQuery(accountEmailAddress);
			((AppAdminConsole)MyApplication).zPageSearchResults.zToolbarPressButton(Button.B_SEARCH);

			// Get the list of displayed accounts
			List<AccountItem> accounts = ((AppAdminConsole)MyApplication).zPageSearchResults.zListGetAccounts();

			for (AccountItem a : accounts) {
				logger.info("Looking for account " + accountEmailAddress + " found: " + a.getGEmailAddress());
				if (accountEmailAddress.equals(a.getGEmailAddress())) {
					accountFound = true;
					break;
				}
			}
			ZAssert.assertNotNull(accountFound, "Verify account is found in search results");
			
			accountLocator = "css=td[id^='SEARCH_MANAGE_data_emailaddress']:contains('" + accountEmailAddress + "')";
			zRightClickAt(accountLocator, "");
							
			if (option.equals(Button.O_DELETE)) {
				optionLocator = "css=td[id='zmi__SCHLV__DELETE_title']";
				page = new DialogForDeleteOperation(MyApplication, ((AppAdminConsole) MyApplication).zPageSearchResults);
				
			} else if (option.equals(Button.O_EDIT)) {
				optionLocator = "css=td[id='zmi__SCHLV__EDIT_title']";
				page = new FormEditAccount(MyApplication);
				
			} else if (option.equals(Button.O_CHANGE_PASSWORD)) {
				optionLocator = "css=td[id='zmi__SCHLV__CHNG_PWD_title']";
				page = null;

			} else if (option.equals(Button.O_VIEW_MAIL)) {
				optionLocator = "css=td[id='zmi__SCHLV__VIEW_MAIL_title']";
				page = null;
				
			} else if (option.equals(Button.O_MOVE_ALIAS)) {
				optionLocator = "css=td[id='zmi__SCHLV__MOVE_ALIAS_title']";
				page = null;

			} else if (option.equals(Button.O_INVALIDATE_SESSIONS)) {
				optionLocator = "css=td[id='zmi__SCHLV__EXPIRE_SESSION_title']";
				page = null;

			} else if (option.equals(Button.O_MOVE_MAILBOX)) {
				optionLocator = "css=td[id='zmi__SCHLV__UNKNOWN_78_title']";
				page = null;
			
			} else if (option.equals(Button.O_SEARCH_MAIL)) {
				optionLocator = "css=td[id='zmi__SCHLV__UNKNOWN_79_title']";
				page = null;
			}

			sClick(optionLocator);
			
		} else {
			
			// How many items are in the page?
			String rowsLocator = "css=div#zl__ACCT_MANAGE div[id$='__rows'] div[id^='zli__']";
			int count = this.sGetCssCount(rowsLocator);
			int scrollCounter = 50;
			
			logger.debug(myPageName() + " zListGetAccounts: number of accounts: " + count);

			if (count >= 50) {
				
				for (int accountPaging = 1; accountPaging <= 30; accountPaging++) {
					
					String pageCounter = rowsLocator + ":nth-child(" + scrollCounter + ")";
					SleepUtil.sleepVerySmall();
					if (this.sIsElementPresent(pageCounter)) {
						sClickAt(pageCounter, "");
						SleepUtil.sleepVerySmall();
						this.zKeyboard.zTypeKeyEvent(KeyEvent.VK_DOWN);
						scrollCounter = scrollCounter + 20;
					} else {
						break;
					}
				}
				
				accountLocator = "//div[@id='zl__ACCT_MANAGE']//div[starts-with(@id,'zli__DWT')]//td[starts-with(@id,'account_data_emailaddress_')]/nobr[.='" + accountEmailAddress + "']";
				//accountLocator = "css=div[id^='zli__DWT'] td[id^='account_data_emailaddress']:contains('" + account + "')";
				if (sIsElementPresent(accountLocator)) {
					accountFound = true;
				}
				
				if (accountFound.equals(true)) {
					if (action == Action.A_LEFTCLICK) {
						sClickAt(accountLocator, "");
					}
				} else {
					throw new HarnessException("Couldn't find account " + accountEmailAddress + " after virtual paging.");
				}
			}
		}

		SleepUtil.sleepMedium();
		return page;
	}

image

### CODE REVIEW

File: src/java/com/zimbra/qa/selenium/projects/admin/ui/PageManageAccounts.java [hunk 1]
```diff
@@ -139,52 +138,173 @@ public void zNavigateTo(String treeItem) {
...
 			}
-
+			
+			accountLocator = "//div[@id='zl__ACCT_MANAGE']//div[starts-with(@id,'zli__DWT')]//td[starts-with(@id,'account_data_emailaddress_')]/nobr[.='" + accountEmailAddress + "']";
+			//accountLocator = "css=div[id^='zli__DWT'] td[id^='account_data_emailaddress']:contains('" + account + "')";
+			if (sIsElementPresent(accountLocator)) {
+				accountFound = true;
+			}
+			
+			if (accountFound.equals(true)) {
```
Comment (singhjitesh) [inline src/java/com/zimbra/qa/selenium/projects/admin/ui/PageManageAccounts.java]: We can just write if (accountFound) {

Comment (jiteshsojitra) [reply src/java/com/zimbra/qa/selenium/projects/admin/ui/PageManageAccounts.java]: Done.

File: src/java/com/zimbra/qa/selenium/projects/admin/ui/PageManageAccounts.java [hunk 2]
```diff
@@ -139,52 +138,173 @@ public void zNavigateTo(String treeItem) {
...
+			List<AccountItem> accounts = ((AppAdminConsole)MyApplication).zPageSearchResults.zListGetAccounts();
+
+			for (AccountItem a : accounts) {
+				logger.info("Looking for account " + accountEmailAddress + " found: " + a.getGEmailAddress());
+				if (accountEmailAddress.equals(a.getGEmailAddress())) {
+					accountFound = true;
+					break;
+				}
+			}
+			ZAssert.assertNotNull(accountFound, "Verify account is found in search results");
```
Comment (singhjitesh) [inline src/java/com/zimbra/qa/selenium/projects/admin/ui/PageManageAccounts.java]: ZAssert.assertTrue(accountFound, "Verify account is found in search result");

Comment (jiteshsojitra) [reply src/java/com/zimbra/qa/selenium/projects/admin/ui/PageManageAccounts.java]: Made this change too, thanks Jitesh.

**Review by silentsakky — APPROVED** [review]

**Review by singhjitesh — APPROVED** [review]

**Pull request closed without merging**